includes/formFactory/ManageWikiFormFactoryBuilder.php

```diff
@@ -315,6 +323,40 @@ private static function buildDescriptorSettings(
 							'default' => $setList[$name] ?? $set['overridedefault']
 						];
 						break;
+					case 'preferences':
+						$preferences = [];
+						foreach( MediaWikiServices::getInstance()->getUserOptionsLookup()->getOptions( $context->getUser() ) as $preference => $val ) {
```

Comment:
Yes, but that doesn't matter. It only takes the keys of existing preferences not their values, it just needed a value do set it to that.

Reply:
As for `array_keys` I will fix that.

Reply:
But we have a different number of preferences for different users - fundamentally, we can have different arrays for different people who view it. So if User A is 180 preferences available and selects 4 of them, and User B modifies the list of 4 preferences, perhaps one of the 4 preferences they don't have - so they'll see 3 preferences not 4. Therefore when they save it, it will remove 1 of those 4 preferences entirely by mistake because User B doesn't have that one preference

Reply:
Removed the type for now, will submit new PR for it if I can figure out how.